### ISSUE TITLE
YAM crashes after edit a mail

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#470]
**Originally by _opionline@yahoo.de_ on 2013-12-11 14:13:43 +0100**

___
YAM crahes after edit a mail
Steps to reproduce:
1. run YAM
2. click "New" and fill in all needed parts
3. write something in the body
4. wait until the mail is autosaved (the mail is in the drafts folder)
5. click send later (the mail is moved to the outgoing folder)
6. go to the outgoing folder
7. select the just created mail
8. click "edit"
9. write something in the body
10. wait until the mail is autosaved
11. click "send"
12. crash - MuOutput.txt attached

maybe this bug is related to bug #468?

BTW: in earlier versions of YAM it was possible to double click the mail at step 7. and the write window was opened. Now a read window is opened. :-(

**Comment by jens-maus** [comment]
**Originally by _opionline@yahoo.de_ on 2013-12-11 14:14:03 +0100**

___
**Attachment added:** [`MuOutput.txt`](https://yam.ch/attachments/470/MuOutput.txt) (11.0 KiB)

**Comment by jens-maus** [comment]
**Originally by _opionline@yahoo.de_ on 2013-12-11 14:17:59 +0100**

___
forget to say:
Used version (still): YAM 2.9-dev (30.11.2013)
Copyright (C) 2000-2013 YAM Open Source Team [OS3/m68k, r7366]

**Comment by jens-maus** [comment]
**Originally on 2013-12-11 14:22:56 +0100**

___
As  usual:

1. Is it *always* reproducible?
2. please use the latest nightly build first to try to reproduce the problem and report back.

**Comment by tboeckel** [comment]
**Originally on 2013-12-11 14:27:58 +0100**

___
Regarding the double click feature of mails in the outgoing folder, the philosophy has changed here. If you click on "send now" or "send later" the mail is considered to be "complete" or "finished" and therefore moved to the outgoing folder to be sent out (either immediately or later). Such mails are now handled like any other (received) mail and a read window will be opened after a double click.

If you think a mail being edited is not yet finished you shouldn't click on these two buttons but on "save as draft" instead. Mails in the draft folder are considered to be "incomplete" or "unfinished" (hence the name) and double clicking these will open a write window again to let you finish them.

**Comment by jens-maus** [comment]
**Originally by _opionline@yahoo.de_ on 2013-12-11 14:52:25 +0100**

___
1. Yes it is *always* reproducible! Did you try it???

2. here was not much changes in the nightly builds and it should be possible to reproduce the bug for you too.
If you really want logfiles from the latest nightly build you must wait at least until friday because such debug sesson with the debug version need really much time.

**Comment by jens-maus** [comment]
**Originally by _opionline@yahoo.de_ on 2013-12-11 15:01:50 +0100**

___
Replying to thboeckel:
> Regarding the double click feature ...

OK Thore. Thanks for the info and with your desription it makes really sense. :-)

**Comment by tboeckel** [comment]
**Originally on 2013-12-11 20:11:07 +0100**

___
I just tried to reproduce this issue step by step with both the normal and the debug build dated 30.11.2013 and all debug tools running. I am sorry to disappoint you, but for me there is no crash happening.

**Comment by jens-maus** [comment]
**Originally by _opionline@yahoo.de_ on 2013-12-13 02:09:10 +0100**

___
I just try the version from 12.12.2013 on a test directory with a minimal config and got no hit, so the cause must be in my config or addressbook. i'll investigate it and report back later.

BTW: After step 10 i have two mails, one in the outgoing and one in the drafts folder. After clicking "send" one mail is sent and moved to the "sent" folder and one is still into the outgoing folder. That should be changed IMHO.

**Comment by tboeckel** [comment]
**Originally on 2013-12-13 06:27:47 +0100**

___
Sorry, no. The button in the toolbar cannot carry the whole message. The mail list context menu names the corresponding entry "Edit as new" instead of just "Edit". That means that a **new** mail with the same contents will be created for editing and the old mail will be left untouched as it is. On the one hand this is done because mails in the Outgoing folder are considered "finished", as explained earlier already, and are to be sent as they are. On the other hand this additional new mail serves as a kind of "undo" function. Even if you make larger permanent changes to the new mail you still have the unchanged "original". Of course the older mail will remain in the Outgoing folder, because only the edited copy will be sent out and moved to the Sent folder.

Basically it is your task to decide which of the two mails is the one to be Sent. The other one is kept and again it is your task to delete it manually. This way nobody can complain that YAM automatically deletes stuff without user interaction.

**Comment by jens-maus** [comment]
**Originally by _opionline@yahoo.de_ on 2013-12-13 14:09:35 +0100**

___
about the edit i'll add a comment in #472. back to the crash:

Before you do the steps above please do the follow steps:
1. Go to config
2. select the filers page
3. click "+" to add a new filter
4. go to "actions"
5. select "move" don't select a folder
6. click "save"
you should get a entry in the config like this:

"
[Filters]
FI00.Name           = - new -
FI00.Remote         = N
FI00.ApplyToNew     = Y
FI00.ApplyToSent    = N
FI00.ApplyOnReq     = Y
FI00.Combine        = 1
FI00.Field0         = 0
FI00.SubField0      = 0
FI00.CustomField0   = 
FI00.Comparison0    = 0
FI00.Match0         = 
FI00.CaseSens0      = N
FI00.Substring0     = N
FI00.DOSPattern0    = N
FI00.SkipEncrypted0 = N
FI00.Actions        = 16416
FI00.RedirectTo     = 
FI00.ForwardTo      = 
FI00.ReplyFile      = 
FI00.ExecuteCmd     = 
FI00.PlaySound      = 
FI00.MoveToFolderID = 00000000

[Spam filter] 
"

With such entry you should get the crash. At least here i get the crash.

i think the entry "...MoveToFolderID # 00000000" cause the crash. I have some "...MoveToFolderIDxxxxxxxx" entries in my config but no folder is displayed in the field right to the "move" checkmark. i guess the entries come form the problems with the filters with the 2.9 version some weeks ago.
Now i have to check my 87 Filters which cause the crash...

**Comment by jens-maus** [comment]
**Originally on 2014-01-05 00:03:21 +0100**

___
Please try to reproduce with official 2.9 version and report back.

**Comment by jens-maus** [comment]
**Originally by _trac-robot_ on 2014-01-19 05:57:44 +0100**

___
This ticket was closed automatically by the system.  It was previously set to a Pending status and hasn't been updated within 14 days.
